tests/test_stnode.py

```diff
@@ -79,3 +80,18 @@ def test_schema_info():
             },
         }
     }
+
+
+def test_set_pattern_properties():
+    """
+    Regression test for `photmjsr` not being validated because it is a patternProperty
+    """
+
+    # This model uses the `photmjsr` patternProperty
+    mdl = factories.create_wfi_img_photom_ref()
+
+    with pytest.raises(asdf.ValidationError):
+        # This should be invalid because it is not a quantity
+        mdl.phot_table.F062.photmjsr = 3.14
```

Comment:
@nden this part of the test should indicate that your issue has been resolved.

As for setting it via the `__setitem__` interface, @ddavis-stsci does not want that interface to run validation. So I cannot prevent that interface from allowing any type of data to be set.